server/src/testflinger/logs.py

```diff
@@ -139,3 +139,27 @@ def retrieve_log_fragments(
             )
             for fragment in log_collection.find(query).sort("fragment_number")
         )
+
+    def format_logs_as_results(self, job_id: str, result_data: dict) -> dict:
+        """Format logs stored in MongoDB into the result_data structure."""
```

Comment:
Since this function doesn't just return the dictionary but modify the `result_data` argument, could you document the side effect on the docstring? Maybe something like:


```suggestion
        """Format logs stored in MongoDB into the result_data structure.

        :param job_id: The job identifier.
        :param result_data: The original result_data structure. This will be modified.
        :return: The modified result_data structure.
        """
```